src/main/java/StudentManagement.java

```diff
@@ -0,0 +1,47 @@
+import CustomExceptions.StudentIsNotRegisteredException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class StudentManagement {
+    List<Student> studentList = new ArrayList<>();
+
+    public Student findStudentById(int id) throws StudentIsNotRegisteredException {
+        boolean isFound = false;
+        for (Student student: studentList) {
+            if(student.getId() == id) {
+                isFound = true;
+                System.out.println(student);
+                return student;
+            }
+        }
+
+        if (!isFound) {
```

Comment:
You don't need to check `isFound` since if the student was found it the for loop you wouldn't get to this point.